src/main/scala/za/co/absa/ultet/dbitems/extractors/DBTableFromPG.scala

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2023 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.ultet.dbitems.extractors
+
+import za.co.absa.balta.classes.setter.{Params, SetterFnc}
+import za.co.absa.balta.classes.{DBConnection, DBQuerySupport, QueryResultRow}
+import za.co.absa.ultet.dbitems.DBTable
+import za.co.absa.ultet.dbitems.table.{DBTableColumn, DBTableIndex}
+import za.co.absa.ultet.dbitems.table.DBTableIndex.{DBPrimaryKey, DBSecondaryIndex, IndexColumn}
+import za.co.absa.ultet.model.{DatabaseName, SchemaName, UserName}
+import za.co.absa.ultet.model.table.{ColumnName, IndexName, TableName}
+import za.co.absa.ultet.util.ResourceReader
+
+case class DBTableFromPG(databaseName: DatabaseName)(implicit connection: DBConnection) extends DBQuerySupport {
+  def extract(implicit schemaName: SchemaName, tableName: TableName): Option[DBTable] = {
+    implicit val schemaAndTableNameSetters: List[SetterFnc] = Params.add(schemaName.normalized).add(tableName.normalized).setters
+    extractDBTable().map {case (dbTable, hasIndexes) =>
+
+      val columns = extractColumns()
+      val (secondaryIndexes, primaryKey) = if (hasIndexes) extractIndexes() else (List.empty, None)
+      dbTable.copy(columns = columns, primaryKey = primaryKey, indexes = secondaryIndexes.toSet)
+    }
+  }
+
+  private def extractDBTable()(implicit schemaName: SchemaName,
+                               tableName: TableName,
+                               schemaAndTableNameSetters: List[SetterFnc]): Option[(DBTable, Boolean)] = {
+    runQuery(ResourceReader.sql("pg_table_indexes"), schemaAndTableNameSetters){ queryResult =>
+      if (queryResult.isEmpty) {
+        None
+      } else {
+        val row = queryResult.next()
+        val dbTable = DBTable(
+          tableName = tableName,
+          schemaName = schemaName,
+          primaryDBName = databaseName,
+          owner = UserName(row.getString("table_owner").get),
```

Comment:
Constants could be extracted into some object. Won't comment on other occurrences, but this applies to the whole application. Please create a ticket for it.

Reply:
The argument for using constants is maintainability. If you use the same string in multiple places and need to change it, you would have to find and change every occurrence. With a constant, you only need to change it in one place. The "table_owner" string might in fact be used only once at this point, but that's not the case for other hardcoded strings, file names, etc. Adopting the practice of using constants instead of hardcoded strings will spare you from potential problems and make your code more maintainable in the future.

Reply:
Sure in cases, when the value is used multiple times. But here they are used exactly ones, and not expected to be used anywhere else in future.

Reply:
Well, yeah. But they come from different queries. So it's a "false repeat". Would the column name change in one of the queries, it would cause a split.
I had tried to have constants for query column names in one my previous projects. Trust me, just adds overhead and no gain in return.